src/js/common/schemaform/save-in-progress/RoutedSavableReviewPage.jsx

```diff
@@ -86,56 +191,136 @@ class RoutedSavableReviewPage extends React.Component {
   }
 
   render() {
-    const { form, user, location } = this.props;
+    const {
+      chapters,
+      disableSave,
+      form,
+      formContext,
+      location,
+      setValid,
+      user,
+      viewedPages
+    } = this.props;
 
-    const contentAfterButtons = (
+    return (
       <div>
-        <SaveStatus
-          isLoggedIn={user.login.currentlyLoggedIn}
-          showLoginModal={user.login.showModal}
-          toggleLoginModal={this.props.toggleLoginModal}
-          form={form}>
-        </SaveStatus>
-        <SaveFormLink
-          locationPathname={location.pathname}
-          form={form}
-          user={user}
-          saveAndRedirectToReturnUrl={this.props.saveAndRedirectToReturnUrl}
-          toggleLoginModal={this.props.toggleLoginModal}/>
+        <div className="input-section">
+          <div>
+            {chapters.map(chapter => (
+              <ReviewCollapsibleChapter
+                expandedPages={chapter.expandedPages}
+                chapterFormConfig={chapter.formConfig}
+                chapterKey={chapter.name}
+                form={form}
+                formContext={formContext}
+                key={chapter.name}
+                onEdit={this.handleEdit}
+                open={chapter.open}
+                pageKeys={chapter.pageKeys}
+                setData={this.setData}
+                setValid={setValid}
+                showUnviewedPageWarning={chapter.showUnviewedPageWarning}
+                toggleButtonClicked={() => this.handleToggleChapter(chapter)}
+                uploadFile={this.props.uploadFile}
+                viewedPages={viewedPages}/>
```

Comment:
Was able to dumb down this component a bit by pulling the querying stuff into `mapStateToProps`. I think scroll behavior should be consolidated before turning this into a couple of nested stateless functional components (later PR). This clean up will make that task easier.